pip/index.py

```diff
@@ -450,10 +452,15 @@ def find_all_candidates(self, project_name):
             )
 
         # This is an intentional priority ordering
-        return (
+        priority_versions = (
             file_versions + find_links_versions + page_versions +
             dependency_versions
         )
+        if return_sorted:
```

Comment:
So the ``return_sorted`` parameter is almost always going to be called with a static boolean, e.g. ``PackageFinder().find_all_candates(some_project, True)``. That's a bit silly I think, particularly since the only thing it's saving is just wrapping a ``sorted()`` around the return result.

So I think it'd be better to move this so that instead of:

```python
PackageFinder().find_all_candidates("pip", True)
```

it is instead being called as:

```python
sorted(PackageFinder().find_all_candidates("pip"))
```